contracts/serverless-v2/CommonChainJobs.sol

```diff
@@ -0,0 +1,327 @@
+        uint256 indexed chainId,
+        address indexed gatewayOperator
+    );
+
+    function relayJob(
+        bytes memory _signature,
+        uint256 _jobId,
+        uint256 _reqChainId,
+        bytes32 _codehash,
+        bytes memory _codeInputs,
+        uint256 _deadline,
+        uint256 _jobRequestTimestamp,
+        uint8 _sequenceId,
+        address _jobOwner
+    ) external {
+        require(block.timestamp <= _jobRequestTimestamp + relayBufferTime, "RELAY_TIME_OVER");
+        require(!jobs[_jobId].isResourceUnavailable, "JOB_MARKED_ENDED_AS_RESOURCE_UNAVAILABLE");
+        require(_sequenceId == jobs[_jobId].sequenceId + 1, "INVALID_SEQUENCE_ID");
+        require(jobs[_jobId].execStartTime == 0, "JOB_ALREADY_RELAYED");
+        require(gateways.isChainSupported(_reqChainId), "UNSUPPORTED_CHAIN");
+
+        // signature check
+        bytes32 digest = keccak256(
+            abi.encode(
+                _jobId,
+                _reqChainId,
+                _codehash,
+                _codeInputs,
+                _deadline,
+                _jobRequestTimestamp,
+                _sequenceId,
+                _jobOwner
+            )
+        );
+        address signer = digest.recover(_signature);
+
+        gateways.allowOnlyVerified(signer);
+
+        address[] memory selectedNodes = executors.selectExecutors(noOfNodesToSelect);
+        // if no executors are selected, then mark isRosourceAvailable flag of the job and exit
+        if(selectedNodes.length < noOfNodesToSelect) {
+            jobs[_jobId].isResourceUnavailable = true;
+            emit JobResourceUnavailable(_jobId, _reqChainId, _msgSender());
+            return;
+        }
+        selectedExecutors[_jobId] = selectedNodes;
+
+        jobs[_jobId] = Job({
+            reqChainId: _reqChainId,
+            codehash: _codehash,
+            codeInputs: _codeInputs,
+            deadline: _deadline,
+            execStartTime: block.timestamp,
+            jobOwner: _jobOwner,
+            gatewayOperator: _msgSender(),
+            outputCount: 0,
+            sequenceId: _sequenceId,
+            isResourceUnavailable: false
+        });
+
+        emit JobRelayed(_jobId, _reqChainId, _codehash, _codeInputs, _deadline, _jobOwner, _msgSender(), selectedNodes);
+    }
+
+    function submitOutput(
+        bytes memory _signature,
+        uint256 _jobId,
+        bytes memory _output,
+        uint256 _totalTime,
+        uint8 _errorCode
+    ) external {
+        require(
+            block.timestamp <= jobs[_jobId].execStartTime + jobs[_jobId].deadline + executionBufferTime, 
+            "EXECUTION_TIME_OVER"
+        );
+
+        // signature check
+        bytes32 digest = keccak256(
+            abi.encode(_jobId, _output, _totalTime, _errorCode)
+        );
+        address signer = digest.recover(_signature);
+
+        executors.allowOnlyVerified(signer);
+
+        require(isJobExecutor(_jobId, signer), "NOT_SELECTED_EXECUTOR");
+
```

Comment:
add a condition that same executor can't submit the jobOutput twice to earn the first and second reward.